### PR TITLE
remove trailing slashes from URL paths automatically

### PR DESCRIPTION
**Pull request by 2is10**
@yipingliao

### CODE REVIEW

**Comment by 2is10** [comment]
This is broadly useful, but the use case that prompted it for me was deleting a library slug in the location bar to get to the user profile.

File: kifi-backend/modules/shoebox/angular/src/route.js [hunk 1]
```diff
@@ -11,7 +11,14 @@ angular.module('kifi')
     $httpProvider.defaults.withCredentials = true;
 
     // URL redirects.
+    var trailingSlashRe = /^([^?#]+)\/([?#].*|$)/;
     $urlRouterProvider
+      .rule(function ($injector, $location) {
+        var match = $location.url().match(trailingSlashRe);
+        if (match) {
+          return match[1] + match[2]; // remove trailing slash
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/route.js:19]: I am curious about what `match[2]` is. Testing it against a string with a trailing slash gives me an empty string for `match[2]`.

**Comment by yipingliao** [comment]
Nice!

File: kifi-backend/modules/shoebox/angular/src/route.js (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/route.js:19]: It can be a query string and/or hash fragment.

Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/route.js:19]: Thanks for the explanation.

**Pull request merged**